Can't compile (zlib) with macOS15.5 (clang17)

On my machine (macOS 15.5, Xcode 16.4), I can't compile `zlib` as the compiler (clang 17) now complains about a 

```
_stdio.h:314:7: error: expected identifier or '(' 
```

It can be fixed by passing `-fno-define-target-os-macros` but ideally this should be conditioned to Clang 17+ ; It might be a problem on earlier clang…

--- a/kiwixbuild/dependencies/zlib.py
+++ b/kiwixbuild/dependencies/zlib.py
@@ -24,4 +24,8 @@ class zlib(Dependency):
         archives = [src_archive, meson_patch]
         #patches = ['zlib_std_libname.patch']
 
-    Builder = MesonBuilder
+    class Builder(MesonBuilder):
+        @property
+        def configure_options(self):
+            if self.buildEnv.configInfo.build == "x86-64_musl":
+                yield "-Dtests=disabled"

--- a/kiwixbuild/dependencies/zlib.py
+++ b/kiwixbuild/dependencies/zlib.py
@@ -14,13 +14,13 @@ class zlib(Dependency):
 
     class Source(ReleaseDownload):
         src_archive = Remotefile(
-            "zlib-1.2.12.tar.gz",
-            "91844808532e5ce316b3c010929493c0244f3d37593afd6de04f71821d5136d9"
+            "zlib-1.3.1.tar.gz", # https://zlib.net/zlib-1.3.1.tar.gz
+            "9a93b2b7dfdac77ceba5a558a580e74667dd6fede4585b91eefb60f03b72df23"
         )
         meson_patch = Remotefile(
-            "zlib_1.2.12-1_patch.zip",
-            "8ec8344f3fe7b06ad4be768fd416694bc56cb4545ce78b0f1c18b3e72b3ec936",
-            "https://wrapdb.mesonbuild.com/v2/zlib_1.2.12-1/get_patch")
+            "zlib_1.3.1-1_patch.zip", # https://wrapdb.mesonbuild.com/v2/zlib_1.3.1-1/get_patch
+            "e79b98eb24a75392009cec6f99ca5cdca9881ff20bfa174e8b8926d5c7a47095"
+        )
         archives = [src_archive, meson_patch]
         #patches = ['zlib_std_libname.patch']
 

--- a/kiwixbuild/versions.py
+++ b/kiwixbuild/versions.py
@@ -33,10 +33,10 @@ release_versions = {
 
 # This is the "version" of the whole base_deps_versions dict.
 # Change this when you change base_deps_versions.
-base_deps_meta_version = "14"
+base_deps_meta_version = "15"
 
 base_deps_versions = {
-    "zlib": "1.2.12",
+    "zlib": "1.3.1",
     "lzma": "5.2.6",
     "zstd": "1.5.2",
     "docoptcpp": "0.6.2",

